alibi_detect/od/sklearn/svm.py

```diff
@@ -0,0 +1,143 @@
+        self,
+        x_ref: np.ndarray,
+        nu: float = 0.5,
+        tol: float = 1e-6,
+        max_iter: int = 1000,
+        verbose: int = 0,
+    ) -> Dict:
+        """Fit the SKLearn SVM model.
+
+        Parameters
+        ----------
+        x_ref
+            Training data.
+        nu
+            The proportion of the training data that should be considered outliers. Note that this does
+            not necessarily correspond to the false positive rate on test data, which is still defined when
+            calling the `infer_threshold` method.
+        tol
+            The decrease in loss required over the previous ``n_iter_no_change`` iterations in order to
+            continue optimizing.
+        max_iter
+            The maximum number of optimization steps.
+        verbose
+            Verbosity level during training. ``0`` is silent, ``1`` a progress bar.
+
+        Returns
+        -------
+        Dictionary with fit results. The dictionary contains the following keys:
+            - converged: `bool` indicating whether training converged.
+            - n_iter: number of iterations performed.
+        """
+        n_components = self.n_components if self.n_components is not None else len(x_ref)
+        self.nystroem = Nystroem(
+            kernel=self.kernel,
+            n_components=n_components,
+            gamma=1. / (2. * self.sigma ** 2) if self.sigma is not None else None,
+            kernel_params=self.kernel_params,
+        )
+        x_ref = self.nystroem.fit(x_ref).transform(x_ref)
+        self.gmm = SGDOneClassSVM(
+            tol=tol,
+            max_iter=max_iter,
+            verbose=verbose,
+            nu=nu
+        )
+        self.gmm = self.gmm.fit(x_ref)
+        self._set_fitted()
+        return {
+            'converged': self.gmm.n_iter_ < max_iter,
+            'n_iter': self.gmm.n_iter_,
+        }
+
+    def format_fit_kwargs(self, fit_kwargs: Dict) -> Dict:
+        """Format kwargs for `fit` method.
+
+        Parameters
+        ----------
+        kwargs
+            dictionary of Kwargs to format. See `fit` method for details.
+
+        Returns
+        -------
+        Formatted kwargs.
+        """
+        return dict(
+            nu=fit_kwargs.get('nu', 0.5),
+            tol=fit_kwargs.get('tol', 1e-3),
+            max_iter=fit_kwargs.get('max_iter', 1000),
+            verbose=fit_kwargs.get('verbose', 0),
+        )
+
+    def score(self, x: np.ndarray) -> np.ndarray:
+        """Computes the score of `x`
+
+        Parameters
+        ----------
+        x
+            `np.ndarray` with leading batch dimension.
+
+        Returns
+        -------
+        `np.ndarray` of scores with leading batch dimension.
+
+        Raises
+        ------
+        NotFittedError
+            Raised if method called and detector has not been fit.
+        """
+        self.check_fitted()
+        x = self.nystroem.transform(x)
+        return - self.gmm.score_samples(x)
```

Comment:
nit: remove space after the minus sign (assuming it's not a bug to have a minus sign!)

Reply:
👍🏻